torch/csrc/autograd/init.cpp

```diff
@@ -36,7 +36,11 @@ template <> struct type_caster<torch::autograd::profiler::EventKind> {
       case torch::autograd::profiler::EventKind::Mark:
         return py::cast("mark").release();
     }
+#ifdef _MSC_VER
+    __assume(0);
+#else
     __builtin_unreachable();
+#endif
```

Comment:
Can you make a macro `ENSURE_UNREACHABLE` that expands to different things depending on the compiler?